play.pokemonshowdown.com/src/client-main.ts

```diff
@@ -373,6 +387,69 @@ class PSTeams extends PSStreamModel<'team' | 'format'> {
 			key: '',
 		};
 	}
+	loadRemoteTeams() {
+		PSLoginServer.query('getteams', {}).then(data => {
+			if (!data) return;
+			if (data.actionerror) {
+				return PS.alert('Error loading uploaded teams: ' + data.actionerror);
+			}
+			for (const team of data.teams as ((Team & { team: string })[])) {
+				let matched = false;
+				for (const curTeam of this.list) {
+					let match = this.compareTeams(team, curTeam);
+					if (match === true) {
+						// prioritize locally saved teams over remote
+						// as so to not overwrite changes
+						matched = true;
+						break;
+					}
+					if (match === 'rename') {
+						delete curTeam.teamid;
+						if (!team.name.endsWith(' (server version)')) {
+							team.name += ' (server version)';
+						}
+					}
+				}
+				team.loaded = false;
+				if (!matched) {
+					// hack so that it shows up in the format selector list
+					team.folder = '';
+					// team comes down from loginserver as comma-separated list of mons
+					// to save bandwidth
+					let mons = team.team.split(',').map((m: string) => ({ species: m, moves: [] }));
+					team.packedTeam = PSTeambuilder.packTeam(mons);
+					team.iconCache = null;
+					this.push(team);
+				}
+			}
+		});
+	}
+	compareTeams(serverTeam: Team & { team: string }, localTeam: Team) {
+		// if titles match exactly and mons are the same, assume they're the same team
+		// if they don't match, it might be edited, but we'll go ahead and add it to the user's
+		// teambuilder since they may want that old version around. just go ahead and edit the name
+		const mons = serverTeam.team.split(',');
+		let matches = 0;
+		const otherMons = PSTeambuilder.unpackTeam(localTeam.packedTeam);
+		for (const mon of mons) {
+			for (const otherMon of otherMons) {
```

Comment:
This is also O(n^2). Can't you just compare the exact strings? Or are you trying to do something like "same species good enough"?

Reply:
In which case both teams can be mapped to species and then sorted and then compared.

Reply:
"Same species, good enough" is generally the approach, yes.

Reply:
Okay. In which case yeah, just map to species, sort, join, and compare.